core/lib/jobs/jobs/jobs.py

```diff
@@ -213,3 +217,30 @@ def _get_placeholders_str(num_placeholders, start=1):
     if start < 1:
         raise ValueError("Initial placeholder value must be >= 1")
     return ", ".join(f"${i}" for i in range(start, start + num_placeholders))
+
+
+async def check_customer_quota(con, customer_id, service) -> Dict[str, bool]:
+    """Query DB for service-specific customer account usage.
+
+    Will be called for all account tiers, unlimited has a value of -1.
+    Returns:
+        - Dictionary containing boolean values for if uploads or job quotas have been reached
+    """
+
+    async with con.transaction():
+        # get service specific usage restrictions for the customer account
+        usage_query = "select usage_restrictions->$1 as usage from customers where id=$2"
+        usage_json = await con.fetchrow(usage_query, service, customer_id)
+        usage_dict = json.loads(usage_json["usage"])
+        # grab total uploads and jobs of customer for specific service
+        jobs_type = "mantarray"  # REMEMBER TO CHANGE BACK AFTER MIGRATING TO USE PULSE3D
+        query = "select count(*) from jobs_result where customer_id=$1 and type=$2 group by upload_id"
+        customer_data = [row["count"] async for row in con.cursor(query, customer_id, jobs_type)]
+        total_uploads = len(customer_data)
+        total_jobs = sum(customer_data)
```

Comment:
Not sure how important this is, but I think you could calculate these two values in the sql query

Reply:
I'll look into it, this will likely be changed soon when we go into a rolling monthly quota